src/emulator/modules/SceKernelThreadMgr/SceThreadmgrCoredumpTime.cpp

```diff
@@ -20,10 +20,20 @@
 EXPORT(int, sceKernelExitThread, int status) {
     const ThreadStatePtr thread = lock_and_find(thread_id, host.kernel.threads, host.kernel.mutex);
     const std::unique_lock<std::mutex> lock(thread->mutex);
-
+    
     thread->to_do = ThreadToDo::exit;
     stop(*thread->cpu);
     thread->something_to_do.notify_all();
+    
+    std::vector<ThreadStatePtr>::iterator thd;
+    for (thd = thread->waiting_threads.begin(); thd != thread->waiting_threads.end(); thd++){
```

Comment:
You can do:

1) `for (auto thd =`... and avoid the iterator declaration above.
2) Better yet, use range-based for loop and avoid line below too:
```cpp
for (auto t : thread->waiting_threads) {
```